features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreViewModel.kt

```diff
@@ -740,4 +739,49 @@ class ExploreViewModel @Inject constructor(
             analyticsService.logEvent(AnalyticsConstants.ExploreDash.SELECT_ATM_MARKER, bundleOf())
         }
     }
+
+    fun trackFilterEvents(
+        dashPaymentOn: Boolean,
+        giftCardPaymentOn: Boolean,
+        isDialogFullyExpanded: Boolean
+    ) {
+        if (dashPaymentOn){
+            analyticsService.logEvent(AnalyticsConstants.ExploreDash.FILTER_SELECT_DASH, bundleOf())
+        }
+        if (giftCardPaymentOn){
+            analyticsService.logEvent(AnalyticsConstants.ExploreDash.FILTER_SELECT_GIFT_CARD, bundleOf())
+        }
+        analyticsService.logEvent(
+            if (sortByDistance == ExploreViewModel.DEFAULT_SORT_BY_DISTANCE)
+            AnalyticsConstants.ExploreDash.FILTER_SORT_BY_DISTANCE else
+                AnalyticsConstants.ExploreDash.FILTER_SORT_BY_NAME,
+            bundleOf())
+        analyticsService.logEvent(
+            if ( _selectedTerritory.value.isEmpty()) AnalyticsConstants.ExploreDash.FILTER_CURRENT_LOCATION else
+                AnalyticsConstants.ExploreDash.FILTER_SELECTED_LOCATION,
+            bundleOf()
+        )
+        analyticsService.logEvent(
+            when(_selectedRadiusOption.value){
+                1 -> AnalyticsConstants.ExploreDash.FILTER_ONE_MILE
+                5 -> AnalyticsConstants.ExploreDash.FILTER_FIVE_MILE
+                50 -> AnalyticsConstants.ExploreDash.FILTER_FIFTY_MILE
+                else -> AnalyticsConstants.ExploreDash.FILTER_TWENTY_MILE
+            }, bundleOf()
+        )
+        analyticsService.logEvent(
+            if (_isLocationEnabled.value == true) AnalyticsConstants.ExploreDash.FILTER_LOCATION_ALLOWED else
+                AnalyticsConstants.ExploreDash.FILTER_LOCATION_DENIED,
+            bundleOf()
+        )
+        analyticsService.logEvent(AnalyticsConstants.ExploreDash.FILTER_APPLY_ACTION, bundleOf())
+        analyticsService.logEvent(
+            if (isDialogFullyExpanded)
+            AnalyticsConstants.ExploreDash.FILTER_SWIPE_DOWN_ACTION_OFF else
+                AnalyticsConstants.ExploreDash.FILTER_SWIPE_DOWN_ACTION_ON, bundleOf())
```

Comment:
What is the purpose of these events and `isDialogFullyExpanded` value? As I understood from the story, the `Swipe Down` AC is simply for tracking the filter dialog dismissal. So it's the same as `FILTER_CANCEL_ACTION` but for dismissing.

Reply:
@ClaudeHangui judging by Brian's comments, we just need 1 event when the dialog is closed by dismissing. Thus it's not clear why do you need 2 events here. You probably don't even need to track `onSlide` as well. Just have a variable that is set when the user clicked on cancel, and then track the "dismissed" event. Variable set  = `explore_filter_cancel_action`, not set = `explore_filter_dismissed`